src/vs/workbench/contrib/terminal/browser/terminalService.ts

```diff
@@ -262,47 +260,45 @@ export class TerminalService implements ITerminalService {
 	}
 
 	handleNewRegisteredBackend(backend: ITerminalBackend) {
-		if (backend.remoteAuthority === this._environmentService.remoteAuthority) {
-			this._primaryBackend = backend;
```

Comment:
I'm pretty sure removing this if will introduce some bugs, this sets the primary backend potentially to secondary ones (eg. local when on a remote window) and breaks `_primaryBackendRegistered`.

Reply:
`_primaryBackendRegistered` was only used by the terminal view. i'll make a build to verify that it doesn't break remote, but local works as expected

Reply:
in fact, I should remove `_primaryBackendRegistered` as it's not used anymore

Reply:
`_primaryBackend` is used in a lot of places though

Reply:
right, didn't remove that part just moved it -

https://github.com/microsoft/vscode/pull/162953/files#diff-f1883fa148c75292499e7150b63a75610d76d58bb20cefff0213d1f5ec6b88fcR258

Reply:
We shouldn't be relying on the order in which they're registered, maybe just wrapping it in an if is enough, I'm not sure though. This stuff is a little fragile

```ts
if (backend.remoteAuthority === this._environmentService.remoteAuthority) {
	this._primaryBackend = backend;
}
```

Reply:
tested on WSL and still works. yeah that seems good to add